eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +119,117 @@ function Confirm-ChangeLogEntry {
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status="(Unreleased)",
+    [String]$Title,
+    [String[]]$Content
+  )
+
+  # Validate relase Status
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    try {
+      $Status = ([System.DateTime]::ParseExact($Status, $dateFormat, $provider)).ToString($dateFormat)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$dateFormat' or use '(Unreleased)'"
+    }
+  }
+
+  $Status = "($Status)"
+  if (!$Content) { $Content = @() }
+  if (!$Title) { $Title = "## $Version $Status" }
+
+  $newChangeLogEntry = [pscustomobject]@{ 
+    ReleaseVersion = $Version
+    ReleaseStatus  = $Status
+    ReleaseTitle   = $Title
+    ReleaseContent = $Content
+  }
+
+  return $newChangeLogEntry
+}
+
+function Add-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $false)]
+    [Hashtable]$ChangeLogEntries,
+    [Parameter(Mandatory = $true)]
+    [String]$ChangeLogLocation,
+    [Parameter(Mandatory = $true)]
+    [String]$NewEntryVersion,
+    [String]$NewEntryReleaseStatus="(Unreleased)",
+    [String[]]$NewEntryContent=""
+  )
+
+  if ($ChangeLogEntries.Contains($NewEntryVersion))
+  {
+    LogError "This Changelog already contains this entry. Do you mean to edit the entry?"
+    return
+  }
+
+  $newChangeLogEntry = New-ChangeLogEntry -Version $NewEntryVersion -Status $NewEntryReleaseStatus `
+  -Content $NewEntryContent
+  $ChangeLogEntries.Add($NewEntryVersion,$newChangeLogEntry)
+
+  $ChangeLogEntries
+  
+  Set-ChangeLogContent -ChangeLogLocation $ChangeLogLocation -ChangeLogEntries $ChangeLogEntries
+}
+
+function Edit-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [Hashtable]$ChangeLogEntries,
+    [Parameter(Mandatory = $true)]
+    [String]$ChangeLogLocation,
+    [Parameter(Mandatory = $true)]
+    [string]$VersionToEdit,
+    [String]$NewEntryReleaseVersion,
+    [String]$NewEntryReleaseStatus,
+    [String[]]$NewEntryReleaseContent
+  )
+
+  if ($NewEntryReleaseVersion) { $ChangeLogEntries[$VersionToEdit].ReleaseVersion = $NewEntryReleaseVersion }
+  if ($NewEntryReleaseStatus) { $ChangeLogEntries[$VersionToEdit].ReleaseStatus = $NewEntryReleaseStatus }
+  if ($NewEntryReleaseContent) { $ChangeLogEntries[$VersionToEdit].ReleaseContent = $NewEntryReleaseContent }
+
+  $ChangeLogEntries
+
+  Set-ChangeLogContent -ChangeLogLocation $ChangeLogLocation -ChangeLogEntries $ChangeLogEntries
+}
+
+function Set-ChangeLogContent {
+  param (
+    [Parameter(Mandatory = $true)]
+    [String]$ChangeLogLocation,
+    [Parameter(Mandatory = $true)]
+    $ChangeLogEntries
+  )
+
+  try {
```

Comment:
Is there any harm in always just importing SemVer.ps1? I think even if you import it more than once it doesn't hurt anything.